src/lib.rs

```diff
@@ -8,6 +11,23 @@ const RESET: &str = "\u{001B}[0m";
 #[cfg(feature = "colors")]
 const RED: &str = "\u{001B}[31m";
 
+#[cfg(feature = "print-defmt")]
+macro_rules! print {
+    (" ") => {
+        // Do nothing if the string is just a space
+    };
+    ($($arg:tt)*) => {
+        defmt::error!($($arg)*);
+    };
+}
+
+#[cfg(not(feature = "print-defmt"))]
+macro_rules! print {
```

Comment:
Maybe we should rename this to `println` since for esp-println it includes the new-line and for defmt it will be its own log message

Reply:
Done.